File: 88.cpp
#include <vector>

using namespace std;

class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        int p1 = m-1, p2 = n-1;
        while(p1>=0&&p2>=0){
        	if(nums1[p1] > nums2[p2]) {
        		nums1[p1+p2+1] = nums1[p1];
        		p1--;
        	}else {
        		nums1[p1+p2+1] = nums2[p2];
        		p2--;
        	}
        }
        if(p2<0) return;
        if(p1<0) {
        	for(int i = 0; i <= p2; ++i) nums1[i] = nums2[i];
        } 
    }
}; 

File: 88.py
class Solution:
    def merge(self, nums1, m, nums2, n):
        """
        :type nums1: List[int]
        :type m: int
        :type nums2: List[int]
        :type n: int
        :rtype: void Do not return anything, modify nums1 in-place instead.
        """
        pointer1 = m-1
        pointer2 = n-1
        while pointer1 >= 0 and pointer2 >= 0:
            if nums1[pointer1] > nums2[pointer2]:
                nums1[pointer1 + pointer2 + 1] = nums1[pointer1]
                pointer1 -= 1
            else:
                nums1[pointer1 + pointer2 + 1] = nums2[pointer2]
                pointer2 -= 1
        if pointer2 < 0:
            return
        if pointer1 < 0:
            nums1[0:pointer2 + 1] = nums2[0:pointer2 + 1]